straxen/analyses/daq_waveforms.py

```diff
@@ -0,0 +1,141 @@
+import numba
+import pandas
+import straxen
+import numpy as np
+import strax
+import pymongo
+import typing
+import matplotlib.pyplot as plt
+
+
+@straxen.mini_analysis()
+def daq_plot(context, figsize=(14, 15), lower_panel_height=6, group_by='link', **kwargs):
+    """
+    Plot with peak, records and records sorted by "link" or "ADC ID"
+    """
+    f, axes = plt.subplots(3, 1,
+                           figsize=figsize,
+                           gridspec_kw={'height_ratios': [1, 1, lower_panel_height]})
+
+    plt.sca(axes[0])
+    plt.title('Peaks')
+    context.plot_peaks(**kwargs,
+                       single_figure=False)
+    xlim = plt.xlim()
+    plt.xticks(rotation=0)
+    plt.grid('y')
+
+    plt.sca(axes[1])
+    plt.title('Recods (by channel number)')
+    context.plot_records_matrix(**kwargs,
+                                single_figure=False)
+    plt.xticks(rotation=0)
+    plt.grid('x')
+    plt.xlim(*xlim)
+
+    plt.sca(axes[2])
+    plt.title('Records (by reader, link, crate)')
+    context.plot_records_matrix(**kwargs,
+                                group_by=group_by,
+                                single_figure=False)
+    plt.xlim(*xlim)
+    plt.grid()
+
+
+def _get_daq_config(
+        context: strax.Context,
+        run_id: str,
+        config_name: str = 'daq_config',
+        run_collection: typing.Optional[pymongo.collection.Collection] = None) -> dict:
+    """
+    Query the runs database for the config of the daq during this run.
+    Either use the context of the runs collection.
+    """
+    if not context.storage[0].__class__.__name__ == 'RunDB' and run_collection is None:
+        raise NotImplementedError('Only works with the runsdatabase')
+    if run_collection is None:
+        run_collection = context.storage[0].collection
+    daq_config = run_collection.find_one({"number": int(run_id)},
+                                         projection={config_name: 1}).get(config_name, None)
+    if daq_config is None:
+        raise ValueError(f'Requested {config_name} does not exist')
+    return daq_config
+
+
+def _board_to_host_link(daq_config: dict, board: int, add_crate=True) -> str:
+    """Parse the daq-config to get the host, link and crate"""
+    for bdoc in daq_config['boards']:
+        try:
+            if int(bdoc['board']) == board:
+                res = f"{bdoc['host']}_link{bdoc['link']}"
+                if add_crate:
+                    res += f"_crate{bdoc['crate']}"
+                return res
+        except KeyError:
+            raise ValueError(f'Invalid DAQ config {daq_config} or board {board}')
+    # This happens if the board is not in the channel map which might
+    # happen for very old runs.

```

Comment:
The current map predates the oldest data, I'd rather throw because this would mean there is something going wrong, and an "unknown" would probably be ignored.

Reply:
For example for run 009104 we had this

Reply:
.... how?

Reply:
Boards 749 and 1440 aren't in the daq config

Reply:
Ah, right, we swapped 124 and 144 out

Reply:
I think 9104 are even older since these boards you mention weren't replaced by the aforementioned boards:
https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:dsg:daq:summer2020work#and_144_are_very_questionable_boards_done

Reply:
Originally Junji swapped 1253 and 1163 for 749 and 1440, then later we swapped 124 and 144 for 1253 and 1163. I don't remember the exact dates of when these happened, but the cable map in the rundoc does reflect the changes.